contracts/AelinUpFrontDeal.sol

```diff
@@ -0,0 +1,299 @@
+                        nftCollectionRules[i].collectionAddress,
+                        nftCollectionRules[i].purchaseAmount,
+                        nftCollectionRules[i].purchaseAmountPerToken
+                    );
+                }
+                hasNftList = true;
+            }
+            // if the first address supports 1155, the entire pool only supports 1155
+            else if (NftCheck.supports1155(nftCollectionRules[0].collectionAddress)) {
+                for (uint256 i = 0; i < nftCollectionRules.length; i++) {
+                    require(NftCheck.supports1155(nftCollectionRules[i].collectionAddress), "can only contain 1155");
+                    nftCollectionDetails[nftCollectionRules[i].collectionAddress] = nftCollectionRules[i];
+
+                    for (uint256 j = 0; j < nftCollectionRules[i].tokenIds.length; j++) {
+                        nftId[nftCollectionRules[i].collectionAddress][nftCollectionRules[i].tokenIds[j]] = true;
+                    }
+                    emit PoolWith1155(
+                        nftCollectionRules[i].collectionAddress,
+                        nftCollectionRules[i].purchaseAmount,
+                        nftCollectionRules[i].purchaseAmountPerToken,
+                        nftCollectionRules[i].tokenIds,
+                        nftCollectionRules[i].minTokensEligible
+                    );
+                }
+                hasNftList = true;
+            } else {
+                revert("collection is not compatible");
+            }
+        }
+
+        // deposit underlying token logic
+        // check if the underlyingDealAmount is more than 0,
+        // if yes, transfer it to this contract and store it in `currentDealTokenTotal` and add it
+        if (_underlyingDealTokenAmount > 0) {
+            currentDealTokenTotal += _underlyingDealTokenAmount;
+            IERC20(_dealData.underlyingDealToken).transferFrom(_sponsor, address(this), _underlyingDealTokenAmount);
+            if (_underlyingDealTokenAmount >= _dealData.underlyingDealTokenTotal) {
+                dealDepositComplete = true;
+                emit DealFullyFunded(address(this), block.timestamp, block.timestamp + _dealData.proRataRedemptionPeriod);
+            } else {
+                emit DepositDealToken(_dealData.underlyingDealToken, _sponsor, _underlyingDealTokenAmount);
+            }
+        }
+    }
+
+    function depositDealTokens(uint256 _underlyingDealTokenAmount) public {
+        UpFrontDeal memory _dealData = dealData;
+        uint256 _currentDealTokenTotal = currentDealTokenTotal;
+
+        require(
+            IERC20(_dealData.underlyingDealToken).balanceOf(msg.sender) >= _underlyingDealTokenAmount,
+            "not enough balance"
+        );
+        require(_currentDealTokenTotal < _dealData.maxDealTotalSupply, "already deposited the total");
+
+        IERC20(_dealData.underlyingDealToken).transferFrom(msg.sender, address(this), _underlyingDealTokenAmount);
```

Comment:
again you need to protect against tokens that burn their supply in the transfer method